routes/index.js

```diff
@@ -376,4 +398,51 @@ router.get('/getAddress/(:id)', common.isAuthenticated, (req, res, next) => {
     })
 })
 
+router.get('/profile/:userId', common.isAuthenticated, (req, res, next) => {
+    var userId = req.params.userId;
+    var payload = {};
+    payload.user;
+    var library = [];
+    var sentBooks = 0;
+    var receivedBooks = 0;
+
+    sql.pool.query(selectUserinfo, [userId], (err, result) => {
```

Comment:
You might want to handle the errors consistently through this function and switch these first two potential error points to use the flash and redirect as well. Not a deal breaker but just so the end user would know what is going on in the case these errors are hit